third_party/nvfuser/csrc/ir_base_nodes.h

```diff
@@ -361,6 +365,37 @@ class TORCH_CUDA_CU_API Val : public Statement {
 
   void resolveIndexDtype();
 
+  //! Get vector of log messages for this object
+  auto getLogMessages() const {
+    return log_messages_;
+  }
+
+  //! Write a message to this object's log
+  void log(std::string op_name, std::vector<std::string> arg_strings) {
+#ifdef NDEBUG
+    static bool warned = false;
```

Comment:
There's `TORCH_WARN` and `TORCH_WARN_ONCE`. Not very familiar with them, but you might want to use either of them.